### DISCUSSION TITLE
Workers tuning for scaling tests on a single machine

### DISCUSSION THREAD

**Discussion by saveveltri** [artilleryio/artillery#1959] [Artillery]
Hello, 

I'm trying to run a Playwright test in artillery. 
Test in itself is not flaky and it works fine for 5 (almost 6 virtual users).

How many virtual users (approximately)  can I simultaneously run on a single machine without having the test validity affected?
is there anything I can do to tune workers to increase test performances?

**Comment by hassy** [comment]
hi @saveveltri 👋 Each virtual user is a separate Chromium instance, so the answer really depends on the machine you're running these tests on. We recommend using AWS Fargate for scaling out Playwright tests (you can easily launch thousands or tens of thousands of browsers that way).

**Reply by mtulioalcantara** [reply]
In AWS Fargate how many workers (--count) should I use for a volume of 1000 vus for example?

**Reply by hassy** [reply]
There's no formula @mtulioalcantara, it depends on your tests (i.e. how much CPU and memory they consume). You need to experiment to see how many concurrent VUs a single worker can run without performance of the browsers being affected.

**Reply by mtulioalcantara** [reply]
Thanks. The point is I've been making tests with 1000 VUs in different types of configuration, from 10 (count / workers) x 100 VUs until 100 (count / workers) x 10 VUs and all of them give me around 15 ~ 20% of failed VUs. I am pretty sure my infrastructure is capable of 1000 visitors so why do I get so many failed VUs?

**Reply by hcsincnet-scott** [reply]
@mtulioalcantara I'm running into the same issue. Have you found a way to resolve it?

**Reply by mtulioalcantara** [reply]
@hcsincnet-scott I have. I am not sure how exactly, but it was not any specific configuration on fargate or playwright /artillery config files, but adjusting volumes of my infrastructure itself (running EC2 and RDS in AWS). I tryed a new setup of machines as described in the images below and for the first time I had nice results with zero failed VUsers.
 
![Captura de tela 2023-09-29 144618](https://github.com/artilleryio/artillery/assets/91919474/cbf88fd5-9124-4600-80cd-d67cf46e495c)

And my YML file was:

config:
  target: "https://www.yourwebsite.com"
  engines:
    playwright: {}
  processor: "./test.js"
  phases:
    - arrivalCount: 10
      duration: 60
scenarios:
  - engine: playwright
    flowFunction: "myList"
    flow: []
    
I hope you find your answer.
